dev-guide/test_framework.md

```diff
@@ -0,0 +1,219 @@
+Note the JGCRI project is located at: https://ci.pnnl.gov/jenkins/job/JGCRI/
+
+#### Artifactory
+Artifactory is essentially a file server.  You can upload any arbitrary files to save there, attach meta data and have multiple versions of.  In addition it offers front ends to serve as a local proxy for such services as CRAN (R packages), Docker Hub, PyPi (Python packages), and many more.  
+You can browse the files at: https://artifactory.pnnl.gov and I will note specific locations where we will save artifacts below.
+
+### Docker
+Not as essential to the testing framework but it will come up so I will describe it.  Docker is a set of tools which aims to allow users to package everything required to run their software, including the operating system, inside a minimal client.  A user can subsequently "pull" this software container and run it with a docker client on their local operating system, whatever it may be, and can make certain local resource, such as a folder containing GCAM, available inside the docker container.  Finally Docker provides a "Docker hub" website as a means to publish your containers, not only for other users to run but also extend to make new containers.
+
+#### Pull Request Notifier Plugin for Bitbucket
+This is a plugin to Bitbucket which allows us to signal Jenkins when to kick off tests.  This plug can trigger on any Bitbucket action, such as pull request opened or updated.  It makes available a number of useful meta data about what happened and will send it off to some arbitrary URL.  In addition it allows us to add a new menu option with in the pull request and even configure arbitrary forms to collect user input to send along too.  The full documentation of what is possible is on it's Github page: https://github.com/tomasbjerre/pull-request-notifier-for-bitbucket
+
+#### Generic Webhook Plugin for Jenkins
+This is the counterpart for the "Pull Request Notifier for Bitbucket".  It will listen for notifications, unpack the metadata, and launch the configured test based off of the metadata.
+Documentation, again on it's Github page: https://github.com/jenkinsci/generic-webhook-trigger-plugin
+
+#### Notify Bitbucket Instance Plugin for Jenkins
+A plugin to update the build status (In Progress, Failure, Success) in Bitbucket.  The Jenkins plugin takes care of all of the details for us automatically filling in the name of the test, build number, and link back to Jenkins so a user can watch the progress / look at logs or build artifacts after the fact.  
+The plugin makes use of [Bitbucket Build Status API](https://developer.atlassian.com/server/bitbucket/how-tos/updating-build-status-for-commits/) which I actually need to call directly to add / update the builds status of the jobs submitted on PIC.
```

Comment:
STRUCTURE:  Remove "I" to be "...which actually need..."

These need to be changed throughout.